File: is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        /*
        *   Plan: cycle through s, maintain cur place in s with char,
        *   when char is found in t, inc s 1, ONLY loop through t one time.
        */

        int cur = 0;

        if(s.equals("")) {
            return true;
        }

        for(int i = 0; i < t.length(); i++) {
            if(s.charAt(cur) == t.charAt(i)) {
                cur++;
            }
            if(cur >= s.length()) {
                return true;
            }
        }

        if(cur >= s.length()) {
            return true;
        }


        return false;
    }
}

File: is_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        pos_s = 0

        if s == "":
            return True
        if len(s) == len(t):
            if s == t:
                return True
            else:
                return False

        for i in range(len(t)):
            if s[pos_s] == t[i]:
                pos_s+=1
                if pos_s == len(s):
                    return True

        if pos_s == len(s):
            return True

        return False
    #solution is O(n) where n is length of t as we only cycle through elements of t once